packages/core/src/auth/auth.ts

```diff
@@ -999,6 +1004,67 @@ export class Auth implements AuthService, ConnectionManager {
         )
         delete this._declaredConnections[conn.startUrl]
     }
+
+    // Local variable handles per session displays, e.g. we forgot a CodeCatalyst connection AND
+    // an Explorer only connection. We only want to display once in this case.
+    // However, we don't want to set this at the global state level until a user interacts with the
+    // notification in case they miss it the first time.
+    #separationPromptDisplayed = false
+    #authSeparationDismissKey = 'aws.toolkit.separationPromptDismissed'
+
+    /**
+     * Displays a sign in prompt to the user if they have been logged out of the Toolkit as part of
+     * separating auth sessions between extensions. Pass a command name to open that login
+     * (e.g. codecatalyst vs explorer login). Passing nothing will open a prompt for that last used command name
+     * (or do nothing if no command name has ever been passed), which is useful to redisplay the prompt after
+     * reloads in case a user misses it.
+     *
+     * TODO: Remove after some time.
+     */
+    public async displaySeparationPrompt(cmd?: string) {
```

Comment:
not a blocker: To keep this code more isolated, and to reduce the amount of code in the Auth class, would putting this in its own class be cleaner? It would still live in `auth.ts`

Reply:
updated